dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/intermediate/components/MultiSelectComponent.kt

```diff
@@ -0,0 +1,86 @@
+package org.dataland.frameworktoolbox.intermediate.components
+
+import org.apache.commons.text.StringEscapeUtils.escapeEcmaScript
+import org.dataland.frameworktoolbox.intermediate.FieldNodeParent
+import org.dataland.frameworktoolbox.intermediate.components.support.SelectionOption
+import org.dataland.frameworktoolbox.specific.datamodel.TypeReference
+import org.dataland.frameworktoolbox.specific.datamodel.elements.DataClassBuilder
+import org.dataland.frameworktoolbox.specific.fixturegenerator.elements.FixtureSectionBuilder
+import org.dataland.frameworktoolbox.specific.uploadconfig.elements.SectionUploadConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.SectionConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.getTypescriptFieldAccessor
+import org.dataland.frameworktoolbox.specific.viewconfig.functional.FrameworkDisplayValueLambda
+import org.dataland.frameworktoolbox.utils.typescript.generateTsCodeForSelectOptionsMappingObject
+
+/**
+ * A MultiSelectComponent represents a selection of valid NACE codes
+ */
+open class MultiSelectComponent(
+    identifier: String,
+    parent: FieldNodeParent,
+) : ComponentBase(identifier, parent) {
+
+    var options: MutableSet<SelectionOption> = mutableSetOf()
+
+    override fun generateDefaultDataModel(dataClassBuilder: DataClassBuilder) {
+        dataClassBuilder.addProperty(
+            identifier,
+            documentSupport.getJvmTypeReference(
+                TypeReference("List", isNullable, listOf(TypeReference("String", false))),
```

Comment:
Is there a good reason why we are using a `String` instead of a purpose-built enum like for the `SingleSelectComponent`?

Reply:
Yes, consistency:

Currently across Dataland, all multi-select-fields are handled as "List<String>" in the backend. (Some examples: NaceCodes, Country Codes, etc.).

I thought it would be strange to start using purpose-built enums now for the devtools-parsed multi select fields, but keep the "other" multi select fields in Dataland as List<String>.
Felt like doing it all in one ticket might be better.

But no strong opinion on that.   Want to change it?

Reply:
Were you able to clarify this during your call?

Reply:
Yes. This is an optional task if we ahve time at the end.

Reply:
I created a sub-task to keep track of this.   We can do it in a seperate PR these next days.
Therefore you can close this comment here.